src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    public static final int FIGURE_COUNT = 5;
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureNumber) {
+            case 0:
+                final double radius = random.nextDouble() * 100;
+
+                return new Circle(color, radius);
+            case 1:
+                final double side = random.nextDouble() * 100;
+
+                return new Square(color, side);
+            case 2:
+                final double firstSide = random.nextDouble() * 100;
+                final double secondSide = random.nextDouble() * 100;
```

Comment:
For Rectangle, create separate variables for both sides before passing them to the constructor.